[DOCS] Adds plugin API changes doc (#122388)
elastic/kibana: docs/developer/plugin-api-changes/plugin-api-changes.asciidoc

[[plugin-api-changes]]
== Plugin API changes in {minor-version}
++++
<titleabbrev>Plugin API changes</titleabbrev>
++++

This page discusses the plugin API changes that you need to be aware of when migrating
your application to {kib} {minor-version}.

> Other versions: {kibana-ref-all}/7.17/plugin-api-changes.html[7.17] |

Reviewer: In the release notes, I removed the 7.x branches since this doc is 8.0-specific.

{kibana-ref-all}/7.12/plugin-api-changes.html[7.12] |
{kibana-ref-all}/7.11/plugin-api-changes-7-11.html[7.11] |
{kibana-ref-all}/7.10/breaking-changes-7.10.html#general-plugin-API-changes-7-10[7.10] |
{kibana-ref-all}/7.9/breaking-changes-7.9.html#general-plugin-API-changes-79[7.9] |
{kibana-ref-all}/7.8/breaking-changes-7.8.html#general-plugin-API-changes-78[7.8]

[[breaking_plugin_v8.0.0_113950]]
.Client upgraded to v8.0
[%collapsible]
====

{es} service provides the new version of the `elasticsearch-js` library.
https://github.com/elastic/elasticsearch-js/issues/1542[This issue]
lists all changes, including the breaking changes.

Refer to https://github.com/elastic/kibana/pull/113950[#113950]

====

[[breaking_plugin_v8.0.0_112505]]
.Support for deprecated `kbn-system-api` header removed
[%collapsible]
====

{kib} removed support for the deprecated `kbn-system-api` header.
Use the `asSystemRequest` flag instead:

```js
const response = await core.http.get({
  path: '/url',
  asSystemRequest: true,
});
```

Refer to https://github.com/elastic/kibana/pull/112505[#112505]

====

[[breaking_plugin_v8.0.0_82643]]
.`IndexPatterns.create` removed
[%collapsible]
====

> `IndexPatterns.create` has been removed in favor of `IndexPatterns.createAndSave`,

Author: ```suggestion
`IndexPatterns.create` has been removed in favor of `IndexPatterns.createAndSave`
```

> to enforce saving index patterns. This is to eliminate the

Author: ```suggestion
to enforce saving index patterns. This eliminates the
```

> creation an index pattern that can't be referenced by other saved objects or {kib} apps.

Reviewer: ```suggestion
creation of an index pattern that can't be referenced by other saved objects or {kib} apps.
```
